packages/image_picker/image_picker/android/src/main/java/io/flutter/plugins/imagepicker/ImagePickerDelegate.java

```diff
@@ -291,21 +279,25 @@ private void launchTakeVideoWithCameraIntent() {
       useFrontCamera(intent);
     }
 
-    boolean canTakePhotos = intentResolver.resolveActivity(intent);
-
-    if (!canTakePhotos) {
-      finishWithError("no_available_camera", "No cameras available for taking pictures.");
-      return;
-    }
-
     File videoFile = createTemporaryWritableVideoFile();
     pendingCameraMediaUri = Uri.parse("file:" + videoFile.getAbsolutePath());
 
     Uri videoUri = fileUriResolver.resolveFileProviderUriForFile(fileProviderName, videoFile);
     intent.putExtra(MediaStore.EXTRA_OUTPUT, videoUri);
     grantUriPermissions(intent, videoUri);
 
-    activity.startActivityForResult(intent, REQUEST_CODE_TAKE_VIDEO_WITH_CAMERA);
+    try {
+      activity.startActivityForResult(intent, REQUEST_CODE_TAKE_VIDEO_WITH_CAMERA);
+    } catch (ActivityNotFoundException e) {
+      try {
+        // If we can't delete the file again here, there's not really anything we can do about it.
+        //noinspection ResultOfMethodCallIgnored
+        videoFile.delete();
```

Comment:
how do we know that `startActivityForResult` deletes the file, or we don't know? Is this specified somewhere?

Reply:
I'm unsure what you are referring to here. The case where `startActivityForResult` succeeds isn't modified in this PR and should be working as before. If it doesn't succeed the tmpfile we just created before trying the call is deleted again. (which theoretically might fail, but if the system doesn't let us delete a file we just created there's nothing we can do about that. :shrug: )

Reply:
I see.  I wasn't sure about the lifecycle of the file